cpp/include/cuvs/neighbors/cagra.hpp

```diff
@@ -644,11 +657,136 @@ auto build(raft::resources const& res,
  */
 
 /**
- * @defgroup cagra_cpp_index_search CAGRA search functions
+ * @defgroup cagra_cpp_index_add_nodes CAGRA graph node addition functions
  * @{
+ * @brief Add new vectors to the index.
+ *
+ * This function expects an input array that contains both the old and new dataset vectors
+ * (new vector appended to the end). This allows explicit control of dataset allocation.
+ * If there are no specific requirements for dataset allocation, then it is recommended
+ * to use cagra::extend to add new vectors.
+ *
+ * See [cagra::extend](#cagra::extend) for usage example
+ *
+ * @tparam T data element type
+ * @tparam IdxT type of the indices
+ *
+ * @param[in] handle raft resources
+ * @param[in] input_updated_dataset_view updated dataset (initial + additional dataset)
+ * @param[in] index CAGRA index
+ * @param[out] updated_graph_view updated graph
```

Comment:
This seems awkward to return to the user- what are we expecting them to do with this parameter? From the user's perspective, this should be getting updated inside the index itself, shouldn't it?